### ISSUE TITLE
openSUSE' check_gcc_output

### ISSUE THREAD

**Issue by DAP-DarkneSS** [colobot/colobot#425]
Hello!

> [  665s] I: Program returns random data in a function
[  665s] E: colobot no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/colobot-colobot-gold-0.1.4-alpha/src/app/pausemanager.cpp:100
[  665s] E: colobot no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/colobot-colobot-gold-0.1.4-alpha/src/object/level/parserline.cpp:49
[  665s] 
[  665s] I: Program returns random data in a function
[  665s] E: colobot no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/colobot-colobot-gold-0.1.4-alpha/src/app/pausemanager.cpp:100
[  665s] E: colobot no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/colobot-colobot-gold-0.1.4-alpha/src/object/level/parserline.cpp:49

Any help please?

PS Add please openSUSE how to install link: http://software.opensuse.org/download.html?project=games&package=colobot

**Comment by krzys-h** [comment]
```
[ 665s] E: colobot no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/colobot-colobot-gold-0.1.4-alpha/src/app/pausemanager.cpp:100
```
I'm pretty sure this must be a false-positive. All possible values of PauseType are handled properly. It might be that with -DASSERTS=OFF it doesn't actually return anything if it gets an invalid value. I'll try to  fix that.

```
[ 665s] E: colobot no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/colobot-colobot-gold-0.1.4-alpha/src/object/level/parserline.cpp:49
```
That's the same problem with assert(false); in a function that's not really used (and won't be since I'm handling this diffrently now). Also will be fixed soon.

And I'll add a link to openSUSE repo to the README.

**Comment by DAP-DarkneSS** [comment]
Thank you!

**Comment by krzys-h** [comment]
It's fixed on the *dev* branch, It'll be merged into master on alpha-0.1.5 release. If you need it working now, add -DASSERTS=ON to "cmake" command (this is enabled by default) and that should cause these warnings to disappear.
The error in parserline.cpp was already fixed while refactoring saving game files.

**Comment by DAP-DarkneSS** [comment]
-DASSERTS=ON seems to change nothing, here is a build log: https://docs.google.com/uc?id=0B0EpD522dPQcVjdEaWZKTEhnYzg&export=download
What commit could I add as a patch except of 0231c2359af275065cbe573e498dafcf1793b6?

**Comment by krzys-h** [comment]
That was fixed in 1477e72ab4fc5ceae3a3c3bba8234fd5e8170367, but I think it's too big for such a simple patch.

Just this should be enough:
```diff
diff --git a/src/object/level/parserline.cpp b/src/object/level/parserline.cpp
index e0e8097..5c32753 100644
--- a/src/object/level/parserline.cpp
+++ b/src/object/level/parserline.cpp
@@ -43,11 +43,6 @@ CLevelParserLine::~CLevelParserLine()
     }
 }
 
-std::string CLevelParserLine::GetLine()
-{
-    assert(false); //TODO
-}
-
 int CLevelParserLine::GetLineNumber()
 {
     return m_lineNumber;
diff --git a/src/object/level/parserline.h b/src/object/level/parserline.h
index 9881ac7..2a4077d 100644
--- a/src/object/level/parserline.h
+++ b/src/object/level/parserline.h
@@ -37,9 +37,6 @@ public:
     CLevelParserLine(std::string command);
     ~CLevelParserLine();
     
-    //! Get line to be saved in level file
-    std::string GetLine();
-    
     //! Get line number
     int GetLineNumber();
     
```

**Comment by DAP-DarkneSS** [comment]
Looks perfect, thank you very much!
